lib/install/validate/storage.go

```diff
@@ -175,3 +177,34 @@ func (v *Validator) suggestDatastore(path string, label string, flag string) {
 		}
 	}
 }
+
+func parseScratchSize(input string) int {
+
+	sizeFormat := regexp.MustCompile(`(\d+)([TMGK])`)
+	var matches []string
+	if matches = sizeFormat.FindStringSubmatch(input); len(matches) != 3 {
+		log.Errorln("Incorrect size format provided for default image size. Should be an integer followed by a unit, such as 10T, 10G, 10M, or 10K")
+		return -1
+	}
+
+	unit := matches[2]
+	var multiplier int
+	switch unit {
+	case "M":
```

Comment:
Didn't want to mess with Math.pow() cause it gives me `float`s which then have to be cast -- if someone sees a more elegant way to do this, though, I'm happy to change it